core/vm/lfvm/interpreter_test.go

```diff
@@ -18,6 +21,392 @@ type example struct {
+			mockStateDB.EXPECT().SetState(ADDRESS_0, HASH_1, HASH_0)
+		}, GAS_START, 106},
+	{"SSTORE Berlin diff value, diff state as db, db same as val, db is not 0 - OK", []Instruction{{PUSH1, 2 << 8}, {PUSH1, 0}, {SSTORE, 0}}, 0, 0, STOPPED, true, false,
+		func(mockStateDB *vm.MockStateDB) {
+			mockStateDB.EXPECT().GetState(ADDRESS_0, HASH_0).Return(HASH_1)
+			mockStateDB.EXPECT().SlotInAccessList(common.Address{0}, HASH_0).Return(true, true)
+			mockStateDB.EXPECT().GetCommittedState(ADDRESS_0, HASH_0).Return(HASH_2)
+			mockStateDB.EXPECT().AddRefund((params.SstoreResetGasEIP2200 - params.ColdSloadCostEIP2929) - params.WarmStorageReadCostEIP2929)
+			mockStateDB.EXPECT().SetState(ADDRESS_0, HASH_0, HASH_2)
+		}, GAS_START, 106},
+}
+
+type OpCodeWithGas struct {
+	OpCode
+	gas uint64
+}
+
+func addOKOpCodes(tests []OpcodeTest) []OpcodeTest {
+	var addedTests []OpcodeTest
+	addedTests = append(addedTests, tests...)
+	for i := PUSH1; i <= PUSH32; i++ {
+		code := []Instruction{{i, 1}}
+		dataNum := int((i - PUSH1) / 2)
+		for j := 0; j < dataNum; j++ {
+			code = append(code, Instruction{DATA, 1})
+		}
+		addedTests = append(addedTests, OpcodeTest{i.String() + " execution", code, 20, 0, STOPPED, false, false, nil, GAS_START, 3})
+	}
+	var opCodes []OpCodeWithGas
+	opCodes = append(opCodes, getInstructionsWithGas(DUP1, SWAP16, 3)...)
+	opCodes = append(opCodes, getInstructionsWithGas(ADD, SUB, 3)...)
+	opCodes = append(opCodes, getInstructionsWithGas(MUL, SMOD, 5)...)
+	opCodes = append(opCodes, getInstructionsWithGas(ADDMOD, MULMOD, 8)...)
+	opCodes = append(opCodes, OpCodeWithGas{EXP, 10})
+	opCodes = append(opCodes, OpCodeWithGas{SIGNEXTEND, 5})
+	opCodes = append(opCodes, OpCodeWithGas{SHA3, 30})
+	opCodes = append(opCodes, getInstructionsWithGas(LT, SAR, 3)...)
+	opCodes = append(opCodes, OpCodeWithGas{SWAP1_POP_SWAP2_SWAP1, 11})
+	opCodes = append(opCodes, OpCodeWithGas{POP_SWAP2_SWAP1_POP, 10})
+	opCodes = append(opCodes, OpCodeWithGas{POP_POP, 4})
+	opCodes = append(opCodes, getInstructionsWithGas(PUSH1_SHL, PUSH1_DUP1, 6)...)
+	//opCodes = append(opCodes, OpCodeWithGas{PUSH2_JUMP, 11})
+	opCodes = append(opCodes, OpCodeWithGas{PUSH2_JUMPI, 13})
+	opCodes = append(opCodes, OpCodeWithGas{PUSH1_PUSH1, 6})
+	opCodes = append(opCodes, OpCodeWithGas{SWAP1_POP, 5})
+	opCodes = append(opCodes, OpCodeWithGas{SWAP2_SWAP1, 6})
+	opCodes = append(opCodes, OpCodeWithGas{SWAP2_POP, 5})
+	opCodes = append(opCodes, OpCodeWithGas{DUP2_MSTORE, 9})
+	opCodes = append(opCodes, OpCodeWithGas{DUP2_LT, 6})
+	for _, opCode := range opCodes {
+		code := []Instruction{{opCode.OpCode, 0}}
+		addedTests = append(addedTests, OpcodeTest{opCode.String() + " OK execution", code, 20, 0, STOPPED, false, false, nil, GAS_START, opCode.gas})
+	}
+	return addedTests
+}
+
+func TestOK(t *testing.T) {
+
+	var mockCtrl *gomock.Controller
+	var mockStateDB *vm.MockStateDB
+
+	tests := addOKOpCodes(opcodeTests)
+
+	for _, test := range tests {
+		testSuccess := true
```

Comment:
Using sub-tests should help simplify the structure here a bit and increase the test runner usability.

Reply:
Done